### DISCUSSION TITLE
lazygit does not work with lazyvim

### DISCUSSION THREAD

**Discussion by theesfeld** [LazyVim/LazyVim#4207] [Q&A] [answered]
Hi!

when I use lazygit from the command line, it works fine, but when I try to use lazygit in lazyvim (space-gg), I get this error any time I try to pull or push:

`"git error:data contained a nul byte that could not be represented as a string"`

lazygit is installed via homebrew

is this a common issue or did I somehow do something wrong?

**Comment by dpetka2001** [comment]
I just pushed something via lazygit in Lazyvim `<leader>gg` and did not have this problem. Must be something specific with the OS you're using. Try deleting `.git/COMMIT_EDITMSG` before pulling/pushing and try again.

**Comment by theesfeld** [comment]
nope, didn't work.
ive tried it in various repos as well.
<img width="720" alt="image" src="https://github.com/user-attachments/assets/534739c6-4036-44a9-901c-0aaf265af498">

nvim -v
using macOS 
NVIM v0.10.1
Build type: Release
LuaJIT 2.1.1720049189

lazygit -v
commit=, build date=, build source=homebrew, version=0.43.1, os=darwin, arch=arm64, git version=2.45.2

lazyvim is up to date

**Reply by dpetka2001** [reply]
Like I said, must be something specific to your system. I don't observe the behavior you're describing at all. It correctly pushes and pulls without any errors on Linux Mint 21.3 (although I don't use it that much because I prefer the git cli). Google to see if there any results for the error you get on your OS and see if any of the results might have a solution.

**Comment by theesfeld** [comment, marked as answer]
you were correct - the issue is with the homebrew version of lazygit. I removed it and installed from the lazygit repo and it works 100%

**Reply by dpetka2001** [reply]
Glad you were able to solve it. Sorry I couldn't be of more help, but I've never had experience with Mac.
